x/staking/src/keeper/redelegation.rs

```diff
@@ -93,39 +110,37 @@ impl<
         ctx: &mut CTX,
         redelegation: &Redelegation,
         completion_time: chrono::DateTime<Utc>,
-    ) -> anyhow::Result<()> {
-        let mut time_slice = self.get_redelegation_queue_time_slice(ctx, completion_time)?;
+    ) {
+        let mut time_slice = self.redelegation_queue_time_slice(ctx, completion_time);
         let dvv_triplet = DvvTriplet {
             del_addr: redelegation.delegator_address.clone(),
             val_src_addr: redelegation.validator_src_address.clone(),
             val_dst_addr: redelegation.validator_dst_address.clone(),
         };
         if time_slice.is_empty() {
-            self.set_redelegation_queue_time_slice(ctx, completion_time, vec![dvv_triplet])?;
+            self.set_redelegation_queue_time_slice(ctx, completion_time, vec![dvv_triplet]);
         } else {
             time_slice.push(dvv_triplet);
-            self.set_redelegation_queue_time_slice(ctx, completion_time, time_slice)?;
+            self.set_redelegation_queue_time_slice(ctx, completion_time, time_slice);
         }
-        Ok(())
     }
 
-    pub fn get_redelegation_queue_time_slice<DB: Database, CTX: QueryableContext<DB, SK>>(
+    pub fn redelegation_queue_time_slice<DB: Database, CTX: QueryableContext<DB, SK>>(
         &self,
         ctx: &mut CTX,
         completion_time: chrono::DateTime<Utc>,
-    ) -> anyhow::Result<Vec<DvvTriplet>> {
+    ) -> Vec<DvvTriplet> {
         let store = ctx.kv_store(&self.store_key);
         let store = store.prefix_store(REDELEGATION_QUEUE_KEY);
 
         let key = completion_time
             .timestamp_nanos_opt()
-            .expect("The timestamp_nanos_opt produces an integer that represents time in nanoseconds.
-                    The error in this method means that some system failure happened and the system cannot continue work.")
+            .expect(TIMESTAMP_NANOS_EXPECT)
             .to_ne_bytes();
         if let Some(bytes) = store.get(&key) {
-            Ok(serde_json::from_slice(&bytes)?)
+            serde_json::from_slice(&bytes).expect(SERDE_ENCODING_DOMAIN_TYPE)
```

Comment:
You should prefer `unwrap_or_corrupt` from `gears::store::database::ext::UnwrapCorrupt;` here. You can't guarantee it won't fail and if it does fail it's probably the database that's corrupt.

Reply:
Used the constant from the module. I can add implementation for `Result<T, E>` to the `UnwrapCorrupt` and use the method instead of constant.